### ISSUE TITLE
Support `Auto` `line_endings` option

### ISSUE THREAD

**Issue by ThatTimothy** [JohnnyMorganz/StyLua#518]
It would be nice to have an option to respect a file's current line endings by not changing it.

This would look line:

```toml
line_endings = "Auto"
```

Basically:
`CRLF` -> `CRLF`
`LF` -> `LF`

**Comment by JohnnyMorganz** [comment]
When I first implemented it, I thought about this option. But then I decided against it, since I thought it would be more reasonable to remain consistent and force one line ending overall.

What is your use case in particular for this? I much prefer the consistency

**Comment by ThatTimothy** [comment]
> What is your use case in particular for this? I much prefer the consistency

I have Git automatically enforce the remote to be `LF` through the `core.autoCRLF` option. This makes it so when you pull from the repo on a Windows machine, `CRLF` is automatically converted to `LF`. On Unix machines, they see no change on their end.

Basically, I just want to disable StyLua from making these changes, as Git handles the consistency for me, externally of StyLua.

I don't really see a reason to not support customization, as the point of something like StyLua is to format how you want it. If someone wanted consistent line endings, they could still set it as so.

I believe, other formatters, such as prettier (which StyLua seems to be inspired by), also support an `Auto` option.

TL;DR I have a use case for it, other formatters support it, and it would cause no breaking changes.

**Comment by LastTalon** [comment]
I'm not sure how I would feel about adding this.

> This allows users to use whatever line ending their programs natively support

All modern editors support LF or CRLF line endings just fine as far as I'm aware. Is there a particular tool in your workflow that doesn't?

> I don't really see a reason to not support customization, as the point of something like StyLua is to format how you want it.

> Other formatters, such as prettier (which StyLua seems to be inspired by), also support an `Auto` option.

StyLua does take great influence from Prettier and part of that means following to some degree their [option philosophy](https://prettier.io/docs/en/option-philosophy.html), which means its meant to be opinionated, rather than allowing everything to be formatted how you want.

Prettier's stance, despite having the option, seems to be the same. That option seems to have been introduced to *allow* users to enable the consistency, since the original behavior was `auto` when it was introduced. Later it was changed to be the current behavior of formatting to LF endings.

From prettier:
> --end-of-line makes it easier for teams to keep CRLFs out of their git repositories.

Prettier has a long history, and they admit they've done things in the past that, with experience and hindsight, they'd no longer do. My worry is that we re-tread the same path without learning from Prettier. Admittedly, StyLua already has some design choices that we may have done differently with more experience, which makes me wary about introducing more.

These are just the thoughts that make me apprehensive about adding this option.

**Comment by ThatTimothy** [comment]
After reading that article and thinking about it for a while, I think you are right. 

Having consistency is more important than having whatever EOL a platform decides to use. Since most modern tooling supports any EOL, I decided to convert my repo to just use LF, so it works everywhere.

Thanks for the insight and steering me away from the lazy option of not wanting to convert things over (which honestly turned out to be easier).
